File: src/main/java/org/jukeboxmc/form/element/Dropdown.java
package org.jukeboxmc.form.element;

import org.jetbrains.annotations.NotNull;
import org.json.simple.JSONArray;
import org.json.simple.JSONObject;
import org.jukeboxmc.form.CustomForm;

import java.util.ArrayList;
import java.util.List;

/**
 * @author GoMint
 * @version 1.0
 */
public class Dropdown extends Element  {

    private final CustomForm form;
    private final List<String> options = new ArrayList<>();
    private int defaultOption = 0;

    public Dropdown( CustomForm customForm, String id, String text ) {
        super( id, text );
        this.form = customForm;
    }

    public @NotNull Dropdown addOption(String option ) {
        this.options.add( option );
        this.form.setDirty();
        return this;
    }

    public @NotNull Dropdown addOption(String option, boolean defaultOption ) {
        if ( defaultOption ) {
            this.defaultOption = this.options.size();
        }

        this.options.add( option );
        this.form.setDirty();
        return this;
    }

    @Override
    public @NotNull JSONObject toJSON() {
        JSONObject obj = super.toJSON();
        obj.put( "type", "dropdown" );

        JSONArray jsonOptions = new JSONArray();
        jsonOptions.addAll( this.options );

        obj.put( "options", jsonOptions );
        obj.put( "default", this.defaultOption );
        return obj;
    }

    @Override
    public Object getAnswer( Object answerOption ) {
        long optionIndex = (long) answerOption;
        this.defaultOption = (int) optionIndex;
        return this.options.get( (int) optionIndex );
    }

}

File: JukeboxMC-API/src/main/kotlin/org/jukeboxmc/api/form/element/Dropdown.kt
package org.jukeboxmc.api.form.element

import com.google.gson.JsonArray
import com.google.gson.JsonObject
import org.jukeboxmc.api.form.CustomForm

/**
 * @author Kaooot
 * @version 1.0
 */
class Dropdown(private val form: CustomForm, id: String, text: String) : Element(id, text) {

    private val options = arrayListOf<String>()
    private var defaultOption = 0

    override fun toJSON(): JsonObject {
        val jsonObject = super.toJSON()
        jsonObject.addProperty("type", "dropdown")
        val options = JsonArray()
        for (option in this.options) {
            options.add(option)
        }
        jsonObject.add("options", options)
        jsonObject.addProperty("default", this.defaultOption)
        return jsonObject
    }

    override fun getAnswer(answerOption: Any): Any {
        val optionIndex = answerOption as Long
        this.defaultOption = optionIndex.toInt()
        return this.options[optionIndex.toInt()]
    }

    fun addOption(option: String): Dropdown {
        this.options.add(option)
        this.form.setDirty()
        return this
    }

    fun addOption(option: String, defaultOption: Boolean): Dropdown {
        if (defaultOption) {
            this.defaultOption = this.options.size
        }
        this.options.add(option)
        this.form.setDirty()
        return this
    }
}